### ISSUE TITLE
Error with background noise augmentation during evaluation

### ISSUE THREAD

**Issue by ilyassmoummad** [DBD-research-group/BirdSet#248]
Hi, first of all congratulations on the awesome work! 

I'm trying to execute (after having downloaded background):

> python3 birdset/eval.py experiment="birdset_neurips24/HSN/LT/efficientnet.yaml"

I have an error with this line:

`datamodule = hydra.utils.instantiate(cfg.datamodule)`

it crashes because of the background noise augmentation :

```
raise EmptyPathException("There are no supported audio files found.")
torch_audiomentations.core.transforms_interface.EmptyPathException: There are no supported audio files found.
```

I have ensured that the dcase18 dataset is downloaded, and in the configuration file, both cfg.datamodule.transforms.waveform_augmentations and background_paths, cfg.paths.background_paths point to the correct path of the background files.

Additionally, I'm wondering why augmentation is needed for evaluation. Thank you very much in advance for your assistance!

**Comment by lurauch** [comment]
Hi Ilyass, 

Thanks for the first (external) issue! :) I'll investigate further and report back with my findings.

Regarding augmentations for evaluation: If we're using a pretrained model and evaluating it on LT or MT, background augmentations should not be necessary. However, for DT, these augmentations are required since we cannot validate on the POW validation dataset (the model does not know the respective classes). As a result, we validate on Xeno-Cano focal recordings, which need augmentations to better resemble a soundscape file. In your case, though, these augmentations should not be needed.

**Comment by lurauch** [comment]
For a quick fix: can you please set the augmentations to none? This should work: 

`python3 birdset/eval.py experiment=birdset_neurips24/HSN/LT/efficientnet.yaml datamodule.transforms.waveform_augmentations=none.yaml datamodule.transforms.spectrogram_augmentations=none.yaml`

We did not test that yet but you can try. You can also add an override statement to the experiment file if you don't want to add everything in the command line.

**Comment by ilyassmoummad** [comment]
Hi Lukas, thanks a lot for your quick reply!

I tried the quick fix but I got this :

```
Error executing job with overrides: ['experiment=birdset_neurips24/HSN/LT/efficientnet.yaml', 'datamodule.transforms.waveform_augmentations=none.yaml']
Error in call to target 'birdset.datamodule.components.transforms.BirdSetTransformsWrapper':
AttributeError("'str' object has no attribute 'get'")
full_key: datamodule.transforms

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

I'm sorry, I don't have experience with complex project management using hydra/yaml, it's probably a newbie error.

**Comment by lurauch** [comment]
Sorry, my bad. Try: 

`python eval.py experiment=birdset_neurips24/HSN/LT/efficientnet.yaml datamodule/transforms/spectrogram_augmentations=none.yaml datamodule/transforms/waveform_augmentations=none.yaml`

This should work. If not, just adjust the complete experiment file: 

```
@package _global_
defaults:

- _self_
- datamodule: HSN
- module: multilabel
- callbacks: default
- paths: default
- trainer: single_gpu
- logger: wandb_default
- hydra: default
- extras: default
- override /datamodule/transforms/spectrogram_augmentations: none.yaml
- override /datamodule/transforms/waveform_augmentations: none.yaml
- experiment: null
- hparams_search: null
- debug: null

task_name: "train"
ckpt_path: null
seed: 1

module:
network: efficientnet

logger:
wandb:
mode: online

train: True
test: True
save_state_dict: True
start_time: ${now:%Y-%m-%d_%H%M%S}
dump_metrics: True
```

**Comment by ilyassmoummad** [comment]
thanks a lot for your help, it works!

**Comment by lurauch** [comment]
We have a solution planned so that the eval.py script does not depend on the availability of augmentations - so thank you for pointing that out! If you have any other questions regarding BirdSet or want to talk about bird research, feel free to open another issue or contact me via mail lukas.rauch.@uni-kassel.de :)

**Comment by lurauch** [comment]
Will be solved with #249
